src/cli/commands/install.js

```diff
@@ -359,6 +359,13 @@ export class Install {
       return true;
     }
 
+    if (!match.integrityMatches && match.integrityError == 'PATTERNS_DONT_MATCH') {
+      this.reporter.warn(this.reporter.lang('integrityPatternsDontMatch'));
+      this.reporter.info(this.reporter.lang('uninstallRegenerate'));
+      this.flags.force = true;
```

Comment:
Okay, should the install command not regenerate the lockfile then, and instead just display a warning in this case?